indexing-engine/index.mdx

```diff
@@ -0,0 +1,97 @@
+
+In the initial release the Indexing Engine provides missing index recommendations, as well as unused index recommendations.
+
+We are also working on index consolidation recommendations in a future release, by using the ["Good Enough Index" Selection](/docs/indexing-engine/good-enough-index-selection) algorithm to help you reduce [Index Write Overhead](/docs/indexing-engine/index-write-overhead).
+
+### Does the Indexing Engine require special extensions?
+
+No. The Indexing Engine is fully built into pganalyze, and as long as you integrate your database with pganalyze (i.e. send query statistics and schema data to pganalyze) no further configuration is necessary.
+
+pganalyze integrates with Postgres running in your own VM, Postgres as a Service running on all leading cloud providers, as well as Postgres-compatible offerings such as Amazon Aurora.
+
+### How do I use the Indexing Engine?
+
+The pganalyze Indexing Engine powers the [pganalyze Index Advisor](/docs/index-advisor) - simply sign up for a pganalyze account, integrate your production system into pganalyze, and you will automatically receive recommendations in your pganalyze account powered by the pganalyze Indexing Engine.
+
+We are also working on making an API available that allows customizing how you use the pganalyze Indexing Engine recommendations in your workflows.
+
+### Is the Indexing Engine using machine learning?
+
+No. The Indexing Engine is powered by a purpose-built optimization algorithm based on the Postgres planner (see architecture overview above), and is executed in a deterministic way. If you provide the same inputs (queries and schema) the same version of the Indexing Engine will always produce the same outputs (recommendations).
+
+Whilst there have been attempts to use machine learning for index selection algorithms, in practice it has been shown (see e.g. ["An Experimental Evaluation of Index Selection Algorithms"](http://www.vldb.org/pvldb/vol13/p2382-kossmann.pdf)) that they take significantly longer to run, without providing better recommendations.
+
+As part of its optimization algorithm, the pganalyze Indexing Engine is able to test possible indexes a lot quicker, due to its unique [What If? analysis](/docs/indexing-engine/what-if-analysis) logic.
+
+### Is the Indexing Engine fully automating index creation for me?
+
+In the initial release the Indexing Engine results require a human review by a database operator. Specifically, when index recommendations are presented through the pganalyze Index Advisor, we expect you to [perform a benchmark on a production copy](/docs/index-advisor/test-opportunities), and if results are suitable, to deploy the index to production with your existing workflows.
+
+In future releases we plan to offer more workflow integrations, for example by automatically testing suggested indexes on a database copy, and, if suitable for your use case, applying them to your production database in a (semi-)automated manner.
```

Comment:
For what it's worth, I don't think it's that useful to talk about future work in this context (other than perhaps allude to it through vague qualifiers like "Currently"), but that's just a gut feeling, so take it or leave it.

Reply:
Yeah, I think that's a fair point. My gut feeling tells me to leave it in for now, to clarify where things are heading, eventually.